### PR TITLE
feat: add new arch

### PR DESCRIPTION
**Pull request by WoLewicki**
## Summary

PR adding new arch to the library. It should no longer be needed to do any autolinking since it is done automatically and TurboModule is used when they are enabled in the project.

## Motivation

It makes the library compatible with new architecture, so users can take advantage of it.

## :pencil: Checklist
- [x] I have performed a self-review of my own code.
- [x] I have optimized code readability (class/variable names, straight forward logic paths, short clarifying docs,...).

## :green_heart: Testing

- [x] I have manually tested my changes. (In Expensify App)

You can run FabricExample app in order to see that it also works correctly on new arch

## Documentation

Select one:
 
- [ ] I have added relevant documentation for my changes.
- [x] This PR does not result in any developer-facing changes. (theoretically since it should be transparent for the users)

### CODE REVIEW

**Comment by WoLewicki** [comment]
@dtroupe-plaid for some reason the newest commits were not applied to the previous PR so here is another one. Could you test if it works correctly?

**Comment by WoLewicki** [comment]
@dtroupe-plaid is there a chance that it can be reviewed and merged somehow?

**Comment by WoLewicki** [comment]
I temporarily added `dist` folder to be able to use it in the project, but I'll delete it when it is time to merge it 🚀

**Comment by dtroupe-plaid** [comment]
> @dtroupe-plaid is there a chance that it can be reviewed and merged somehow?

Yes. Sorry for the delay I was OOO and prior to that we had a packed roadmap. I will have time to look at this tomorrow in the hopes we can merge it next week 🤞 can you resolve any conflicts and merge master (this change will only be supported in V11.x) 🙏

**Review by dtroupe-plaid — COMMENTED** [review]
@WoLewicki I reviewed the PR and don't have many comments on the code. However, I do have some concerns about the example applications. Both examples work on iOS, but I was unable to get either example to run successfully on Android. 

- For the existing example app `npm run android` fails with a compiler error. 
- For the Fabric example the app appears to crash or freeze immediately upon opening. (see attached screen recording). 
- I tested android using a Pixel 7 API 34

Once the issues with the example applications are resolved I can approve this 🙏 

Additionally, the fabric example will need a simple README similar to the [existing example](https://github.com/plaid/react-native-plaid-link-sdk/blob/master/example/README.md) (not a blocker on this PR). 

Let me know if you have any questions or concerns.

File: android/src/main/java/com/plaid/PlaidModule.kt [hunk 1]
```diff
@@ -113,6 +113,26 @@ class PlaidModule internal constructor(reactContext: ReactApplicationContext) :
...
+    Log.e("PlaidModule", "create is not available on Android")
+  }
+
+  override fun open(fullScreen: Boolean, onSuccess: Callback?, onExit: Callback?) {
+    Log.e("PlaidModule", "open is not available on Android")
+  }
+
+  override fun dismiss() {
+    Log.e("PlaidModule", "dismiss is not available on Android")
+  }
```
[the remark this answers is not in the document]
Comment (WoLewicki) [reply android/src/main/java/com/plaid/PlaidModule.kt:130]: We have to implement the methods of the interface

File: ios/RNLinksdk.mm [hunk 2]
```diff
@@ -171,6 +172,14 @@ - (void)stopObserving {
     self.linkHandler = nil;
 }
 
+- (void)continueFromRedirectUriString:(NSString *)redirectUriString { 
+    RCTLogError(@"continueFromRedirectUriString is not supported on iOS");
+}
```
[the remark this answers is not in the document]
Comment (WoLewicki) [reply ios/RNLinksdk.mm:177]: Yeah, I left it since it was present in JS but I can remove it everywhere 🚀

**Comment by WoLewicki** [comment]
@dtroupe-plaid could you check again if everything builds correctly and there are no more crashes? And if so, could we get it merged and released? 🚀

**Comment by WoLewicki** [comment]
I am open to merging the newest `master` branch to resolve conflicts.

**Comment by dtroupe-plaid** [comment]
I should have time to review this next week. If everything works we can merge it 😄

**Comment by WoLewicki** [comment]
@dtroupe-plaid I resolved conflicts and migrated the new view to new arch also. Could you review it and test that the view behaves correctly so we can merge it?

File: android/src/main/java/com/plaid/PLKEmbeddedViewManager.kt [hunk 3]
```diff
@@ -1,10 +1,22 @@
 package com.plaid
 
+import com.facebook.react.module.annotations.ReactModule
 import com.facebook.react.uimanager.SimpleViewManager
 import com.facebook.react.uimanager.ThemedReactContext
+import com.facebook.react.uimanager.ViewManagerDelegate
 import com.facebook.react.uimanager.annotations.ReactProp
+import com.facebook.react.viewmanagers.PLKEmbeddedViewManagerDelegate
+import com.facebook.react.viewmanagers.PLKEmbeddedViewManagerInterface
```
Comment (dtroupe-plaid) [inline android/src/main/java/com/plaid/PLKEmbeddedViewManager.kt:9]: The changes in this file prevent the current example from compiling for Android.

```
file:///Users/dtroupe/plaid/client/link/react-native/react-native-plaid-link-sdk/example/node_modules/react-native-plaid-link-sdk/android/src/main/java/com/plaid/PLKEmbeddedViewManager.kt:8:40 Unresolved reference: PLKEmbeddedViewManagerDelegate
```

```
Failed to install the app. Command failed with exit code 1: ./gradlew app:installDebug -PreactNativeDevServerPort=8081
e: file:///Users/dtroupe/plaid/client/link/react-native/react-native-plaid-link-sdk/example/node_modules/react-native-plaid-link-sdk/android/src/main/java/com/plaid/PLKEmbeddedViewManager.kt:11:3 Unresolved reference: PLKEmbeddedViewManagerInterface
e: file:///Users/dtroupe/plaid/client/link/react-native/react-native-plaid-link-sdk/example/node_modules/react-native-plaid-link-sdk/android/src/main/java/com/plaid/PLKEmbeddedViewManager.kt:15:16 Unresolved reference: PLKEmbeddedViewManagerDelegate
e: file:///Users/dtroupe/plaid/client/link/react-native/react-native-plaid-link-sdk/example/node_modules/react-native-plaid-link-sdk/android/src/main/java/com/plaid/PLKEmbeddedViewManager.kt:27:3 'setToken' overrides nothing
e: file:///Users/dtroupe/plaid/client/link/react-native/react-native-plaid-link-sdk/example/node_modules/react-native-plaid-link-sdk/android/src/main/java/com/plaid/PLKEmbeddedViewManager.kt:31:3 'setIOSPresentationStyle' overrides nothing FAILURE: Build failed with an exception
```

**Review by dtroupe-plaid — APPROVED** [review]
The current Android example still does not work. I can't approve this until the example applications work. 

I apologize for the delay - Please test the Android example app and ensure it works before requesting another review 🙏

**Review by dtroupe-plaid — COMMENTED** [review]
The current Android example app doesn't compile I can't approve this until the example application works. 

I apologize for the delay - I would like to get this merged to help, but I don't get a lot of time allocated to this. Please test the current Android example locally to resolve these issues before requesting another review 🙏 


You can test local changes with yalc like this:

`tsc && yalc publish && cd example && yalc add react-native-plaid-link-sdk && npm install && npm run android`

**Comment by WoLewicki** [comment]
@dtroupe-plaid could you check again? I added missing files.

**Review by dtroupe-plaid — APPROVED** [review]
Retested everything looks 👌

**Pull request merged**